db/flush_job.cc

```diff
@@ -306,6 +317,272 @@ void FlushJob::Cancel() {
   base_->Unref();
 }
 
+Status FlushJob::MemPurge(autovector<MemTable*>& purged_mems) {
+  Status s;
+  db_mutex_->AssertHeld();
+  db_mutex_->Unlock();
+  assert(!mems_.empty());
+
+  // Store the full output memtables in
+  // autovector "purged_mems".
+  // autovector<MemTable*> purged_mems = {};
+  purged_mems = {};
```

Comment:
Nit: for an accumulator output parameter, you have essentially three options:

* assert it's already empty
* just add to what's already there
* blindly replace anything that might be there

The last of these is my least favorite because it seems inherently unsafe from a maintenance perspective.

Reply:
Very true - I just updated the code with with the "assert it's already empty" option.